### PR TITLE
[DataFrame] Implemented cummax, cummin, cumsum, cumprod

### PR DESCRIPTION
**Pull request by devin-petersohn**
@hsubbaraj I tried to rebase for you but ran into problems because you were editing your master branch. I went ahead and submitted a new PR with the same commit history.

### CODE REVIEW

File: python/ray/dataframe/dataframe.py [hunk 1]
```diff
@@ -698,24 +699,127 @@ def cov(self, min_periods=None):
...
+        """Perform a cumulative maximum across the DataFrame.
+
+        Args:
+            axis (int): The axis to take maximum on.
+            skipna (bool): True to skip NA values, false otherwise.
+
+        Returns:
+            The cumulative maximum of the DataFrame.
+        """
+        if(axis == 1):
```
Comment (robertnishihara) [inline python/ray/dataframe/dataframe.py]: There are a couple instances if this that are probably linting errors. Should just be `if axis == 1:`

**Review by robertnishihara — APPROVED** [review]
Looks good to me assuming the linting test passes (on the travis PR build on the main repo).

File: python/ray/dataframe/dataframe.py (hunk 1 above)
Comment (devin-petersohn) [reply python/ray/dataframe/dataframe.py]: Ok, I will fix these.

**Comment by devin-petersohn** [comment]
Test failures unrelated. OK to merge.

**Pull request merged**